shared/js/background/chrome-events.es6.js

```diff
@@ -38,6 +45,118 @@ chrome.runtime.onInstalled.addListener(function (details) {
     }
 })
 
+/**
+ * ONBOARDING
+ * Logic to allow the SERP to display onboarding UI
+ */
+
+/**
+  * !! This function _can_ be sync (`cb` is called in the same tick) if the settings are alrady loaded
+  */
+function getOnboardingParams (cb) {
+    function onReady () {
+        cb(null, {
+            showWelcomeBanner: settings.getSetting('showWelcomeBanner'),
+            showCounterMessaging: settings.getSetting('showCounterMessaging')
+        })
+    }
+
+    if (settings.isReady()) {
+        onReady()
+    } else {
+        settings.ready().then(onReady)
+    }
+}
+
+let onBeforeNavigateTimeStamp = null
+chrome.webNavigation.onBeforeNavigate.addListener(details => {
+    onBeforeNavigateTimeStamp = details.timeStamp
+})
+
+chrome.webNavigation.onCommitted.addListener(details => {
+    getOnboardingParams((err, { showWelcomeBanner, showCounterMessaging }) => {
```

Comment:
Can we remove this arg if not used?

Reply:
Both `showWelcomBanner` and `showCounterMessaging` are used see:

```js
createOnboardingCode({
  isAddressBarQuery,
  showWelcomeBanner,
  showCounterMessaging,
  browser,
  extensionId: chrome.runtime.id
})
```

Reply:
Sorry I wasn't clear, I meant `err`.
I think this is covered by other comments were we can just make it async instead.

Reply:
I have refactored the code to use async/await instead of callbacks